src/Grpc.Net.Client/ChannelBuilder.cs

```diff
@@ -0,0 +1,157 @@
+using System.Diagnostics;
+using System.Net.Http;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+namespace Grpc.Net.Client
+{
+    public class ChannelBuilder
+    {
+        private GrpcChannelOptions _options;
+        private ILoggerFactory? _loggerFactory;
+
+        private ChannelBuilder() : this(new GrpcChannelOptions())
+        {
+        }
+
+        private ChannelBuilder(GrpcChannelOptions options)
+        {
+            _options = options;
+        }
+
+        public static ChannelBuilder ForUrl(Uri uri)
+        {
+            var httpClient = new HttpClient() { BaseAddress = uri };
+
+            return ForHttpClient(httpClient);
+        }
+
+        public static ChannelBuilder ForHttpClient(HttpClient httpClient)
+        {
+            ChannelBuilder channelBuilder = new ChannelBuilder();
+            channelBuilder.ConfigureOptions(o =>
+            {
+                o.HttpClient = httpClient;
+            });
+            return channelBuilder;
+        }
+
+        public static ChannelBuilder ForOptions(GrpcChannelOptions channelOptions)
+        {
+            return new ChannelBuilder(channelOptions);
+        }
+
+        public ChannelBuilder ConfigureOptions(Action<GrpcChannelOptions> configure)
+        {
+            configure(_options);
+            return this;
+        }
+
+        public ChannelBuilder SendMaxMessageSize(int? sendMaxMessageSize)
+        {
+            _options.SendMaxMessageSize = sendMaxMessageSize;
+            return this;
+        }
+
+        public ChannelBuilder ReceiveMaxMessageSize(int? receiveMaxMessageSize)
+        {
+            _options.ReceiveMaxMessageSize = receiveMaxMessageSize;
+            return this;
+        }
+
+        public ChannelBuilder Credentials(ChannelCredentials credentials)
+        {
+            _options.Credentials = credentials;
+            return this;
+        }
+
+        public ChannelBuilder LoggerFactory(ILoggerFactory loggerFactory)
+        {
+            _loggerFactory = loggerFactory;
+            return this;
+        }
+
+        public GrpcChannel Build()
+        {
+            return new GrpcChannel(_options, _loggerFactory ?? NullLoggerFactory.Instance);
+        }
+    }
+
+    public class GrpcChannelOptions
+    {
+        public HttpClient? HttpClient { get; set; }
+
+        public ChannelCredentials? Credentials { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum message size in bytes that can be sent from the client.
+        /// </summary>
+        public int? SendMaxMessageSize { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum message size in bytes that can be received by the client.
+        /// </summary>
+        public int? ReceiveMaxMessageSize { get; set; }
+    }
+
+    public class GrpcChannel : ChannelBase
```

Comment:
GrpcChannel doesn't add anything public that ChannelBase doesn't already have. It is public in case it might need new API in the future.

Reply:
Should be sealed.